### PR TITLE
Pagination - I

### PR DESCRIPTION
**Pull request by Parth858**
Added basic pagination as per Django documentation for jobs,user list view retrieved per page number of data is 2 (for sample) also added dynamic endpoint filters for get_jobs by category and get posted jobs with pagination.

### CODE REVIEW

File: apps/jobs/views.py [hunk 1]
```diff
@@ -72,13 +73,26 @@ def list(self, request):
         except django.core.exceptions.ValidationError as err:
             return response.create_response(err.messages, status.HTTP_404_NOT_FOUND)
         else:
+            # Use Paginator for the queryset
+            page_number = request.GET.get("page", 1)
+            paginator = Paginator(jobs_data, 5)  # 5 items per page
```
Comment (HimanshuSharma-Angel) [inline apps/jobs/views.py:78]: Use it from constant, store it in constant.

File: apps/jobs/views.py [hunk 2]
```diff
@@ -319,38 +334,100 @@ def featured_jobs(self, request):
         """
 
         # past 3 weeks datetime specified for featured jobs (can be modified as per use)
-        past_3_weeks_datetime = datetime_safe.datetime.now(tz=timezone.utc) - timedelta(days=18)
+        past_3_weeks_datetime = datetime_safe.datetime.now(tz=timezone.utc) - timedelta(
+            days=18
+        )
```
Comment (HimanshuSharma-Angel) [inline apps/jobs/views.py:339]: keep it in constant

File: apps/jobs/views.py [hunk 3]
```diff
@@ -319,38 +334,100 @@ def featured_jobs(self, request):
...
-                featured_jobs[0:5],
-                status.HTTP_200_OK
+                response.SOMETHING_WENT_WRONG, status.HTTP_500_INTERNAL_SERVER_ERROR
             )
-        except Exception:
+
+    @action(detail=False, methods=["get"])
+    def get_posted_jobs(self, request):
+        """
+        API: localhost:8000/jobs/get_posted_jobs/
```
Comment (HimanshuSharma-Angel) [inline apps/jobs/views.py:371]: base_url

**Review by HimanshuSharma-Angel — COMMENTED** [review]
make sure to run your changes with different test cases.

File: apps/jobs/views.py [hunk 4]
```diff
@@ -427,6 +428,54 @@ def get_job_categories(self, request):
                 serialized_job_categories, status.HTTP_200_OK
             )
 
+    @action(detail=False, methods=["get"])
+    def get_jobs(self, request):
```
Comment (Parth858) [inline apps/jobs/views.py:432]: As per the FE requirement's updated the code. Added filters which can dynamically retrieve the data according to the request.

1) Retrive jobs as per quary parameter.
```
request: http://localhost:8000/jobs/get_jobs/?category=Security&job_type=full time&qualifications=OWASP&experience=0

response: {
            "job_id": "4332fa40-2c96-43a7-9eae-6978075a3be3",
            "job_role": "Security Analysist",
            "location": "Mumbai",
            "post_date": "2023-10-11",
            "posted": true,
            "experience": 0,
            "created_at": "2024-01-11T18:04:30.138365Z",
            "updated_at": "2024-01-11T18:04:30.138365Z",
            "employer_id": "5e1f2925-9deb-492f-8a06-f451daf9310a",
            "job_type": "full time",
            "salary": "30000.00",
            "qualifications": "OWASP",
            ..... 
          }
```
2) Dynamically retriving of jobs as per the queary updation
```
request: http://localhost:8000/jobs/get_jobs/?job_type=full time&experience=0
response: {
            "job_id": "be5f509a-855a-4de8-b13e-7cb6cf2243fc",
            "job_role": "Product Manager",
            "location": "California",
            "post_date": "2023-10-11",
            "posted": true,
            "experience": 0,
            "created_at": "2024-01-11T17:58:36.746918Z",
            "updated_at": "2024-01-11T17:58:36.746918Z",
            "employer_id": "5e1f2925-9deb-492f-8a06-f451daf9310a",
            "job_type": "full time",
            "salary": "20000.00",
            "qualifications": "Commerces, Data Managment",
            ...
```

**Review by Parth858 — COMMENTED** [review]
As Discussed With FE the code has been updated with filters which can retrieve jobs as per the user preference dynamically.

**Review by HimanshuSharma-Angel — APPROVED** [review]
approving @Himan10 please validate

File: apps/jobs/constants/values.py [hunk 5]
```diff
@@ -32,3 +29,10 @@
 DB_TABLE_COMPANY = "tbl_company"
 DB_TABLE_JOBS = "tbl_job"
 DB_TABLE_USER_PROFILE = "tbl_user_profile"
+
+RESUME_DOCUMENT_TYPE = "resume"
+PROFILE_PICTURE_DOCUMENT_TYPE = "profile_picture"
+COVER_LETTER_DOCUMENT_TYPE = "cover_letter"
+
+ITEMS_PER_PAGE = 5
+PAST_3_WEEK_DATETIME_DAYS18 = 18
```
Comment (Himan10) [inline apps/jobs/constants/values.py:38]: change the variable name to something else, something that resembles to the functionality for which we are using it.

**Review by Himan10 — CHANGES_REQUESTED** [review]

**Comment by Himan10** [comment]
@Parth858 share the latest request/response of the API endpoints present in this PR.

File: apps/jobs/constants/values.py (hunk 5 above)
Comment (Parth858) [reply apps/jobs/constants/values.py:38]: lets not changes the name they are as we discussed and as per the functionality.

**Review by hims1911 — APPROVED** [review]
lgtm @Himan10 you can merge

**Pull request merged**